app/services/valid_test_data_generators/applications_populater.rb

```diff
@@ -106,5 +107,20 @@ def create_declarations(application)
         )
       end
     end
+
+    def create_outcomes(application)
+      completed_declaration = application.declarations.eligible_for_outcomes(lead_provider, application.course.identifier).first
+      return unless completed_declaration
+
+      ParticipantOutcomes::Create::STATES.reverse.each do |state|
+        break if Faker::Boolean.boolean(true_ratio: 0.3)
```

Comment:
I think we should always have outcomes for completed declarations, could we remove this please?

Reply:
Hi @cwrw agree.. I've moved this line to the end of the method so we make sure at least one outcome is created for all completed declarations and some will have 2 outcomes (passed then failed).